src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -1509,4 +1584,142 @@ public Snapshot getSnapshot(String clusterName, String snapshotName) {
+                  .withValue(row.getDouble("value"))
+                  .build());
+        }
+      }
+    }
+
+
+    return metrics;
+  }
+
+  @Override
+  public void storeMetric(GenericMetric metric) {
+    session.execute(
+        storeMetricsPrepStmt.bind(
+            metric.getClusterName(),
+            metric.getMetricDomain(),
+            metric.getMetricType(),
+            metric.getTs().toString(HOURLY_FORMATTER),
+            metric.getHost(),
+            metric.getMetricScope(),
+            metric.getMetricName(),
+            metric.getTs().toDate(),
+            metric.getMetricAttribute(),
+            metric.getValue()));
+  }
+
+  @Override
+  public void storeCompactions(String clusterName, String host, List<Compaction> activeCompactions)
+      throws JsonProcessingException {
+    session.executeAsync(
+        insertOperationsPrepStmt.bind(
+            clusterName,
+            OP_COMPACTION,
+            DateTime.now().toString(HOURLY_FORMATTER),
+            host,
+            DateTime.now().toDate(),
+            objectMapper.writeValueAsString(activeCompactions)));
+  }
+
+  @Override
+  public List<Compaction> listCompactions(String clusterName, String host)
+      throws IOException {
+    // cluster = ? AND type = ? and time_bucket = ? and host = ?
+    ResultSet compactions
+        = session.execute(
+            listOperationsForNodePrepStmt.bind(
+                clusterName, OP_COMPACTION, DateTime.now().toString(HOURLY_FORMATTER), host));
+    return compactions.isExhausted()
+        ? Collections.emptyList()
+        : parseJson(compactions.one().getString("data"), new TypeReference<List<Compaction>>(){});
+  }
+
+  @Override
+  public void storeStreams(String clusterName, String host, Set<CompositeData> activeStreams)
+      throws JsonProcessingException {
+    session.executeAsync(
+        insertOperationsPrepStmt.bind(
+            clusterName,
+            OP_STREAMING,
+            DateTime.now().toString(HOURLY_FORMATTER),
+            host,
+            DateTime.now().toDate(),
+            objectMapper.writeValueAsString(activeStreams)));
+  }
+
+  @Override
+  public Set<CompositeData> listStreamingOperations(String clusterName, String host)
+      throws IOException {
+    // cluster = ? AND type = ? and time_bucket = ? and host = ?
+    ResultSet streams
+        = session.execute(
+            listOperationsForNodePrepStmt.bind(
+                clusterName, OP_STREAMING, DateTime.now().toString(HOURLY_FORMATTER), host));
+    return streams.isExhausted()
+        ? Collections.emptySet()
+        : parseJson(streams.one().getString("data"), new TypeReference<Set<CompositeData>>(){});
+  }
+
+
+  private static <T> T parseJson(String json, TypeReference<T> ref) {
+    try {
+      return new ObjectMapper().readValue(json, ref);
+    } catch (IOException e) {
+      LOG.error("error parsing json", e);
+      throw new RuntimeException(e);
```

Comment:
`RuntimeException` should be treated as an abstract class. this is an `IlegalStateException` (the json state stored is illegal)

Reply:
Sure thing, it's been moved from here anyway as Json parsing shouldn't happen in the DAO.